viz_scripts/ebike_specific_metrics.ipynb

```diff
@@ -144,93 +110,87 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "## Mapping new labels with dictionaries\n",
+    "# Mapping new labels with dictionaries\n",
     "expanded_ct['Trip_purpose']= expanded_ct['purpose_confirm'].map(dic_pur)\n",
     "expanded_ct['Mode_confirm']= expanded_ct['mode_confirm'].map(dic_re)\n",
-    "expanded_ct['Replaced_mode']= expanded_ct['replaced_mode'].map(dic_re)"
+    "expanded_ct['Replaced_mode']= expanded_ct['replaced_mode'].map(dic_re)\n",
+    "\n",
+    "# Mapping fuel\n",
+    "expanded_ct['Mode_confirm_fuel']= expanded_ct['Mode_confirm'].map(dic_fuel)\n",
+    "expanded_ct['Replaced_mode_fuel']= expanded_ct['Replaced_mode'].map(dic_fuel)\n",
+    "\n",
+    "# Change meters to miles\n",
+    "scaffolding.unit_conversions(expanded_ct)\n",
+    "\n",
+    "# Calculate energy impact\n",
+    "expanded_ct = scaffolding.energy_intensity(expanded_ct, df_ei, 'distance_miles', 'Replaced_mode', 'Mode_confirm')\n",
+    "expanded_ct = scaffolding.energy_impact_kWH(expanded_ct, 'distance_miles', 'Replaced_mode', 'Mode_confirm')\n",
+    "expanded_ct = scaffolding.CO2_impact_lb(expanded_ct, 'distance_miles', 'Replaced_mode', 'Mode_confirm')"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "coral-bicycle",
+   "id": "empty-intensity",
    "metadata": {},
    "outputs": [],
    "source": [
-    "dic_pur"
+    "file_suffix = scaffolding.get_file_suffix(year, month, program)\n",
+    "quality_text = scaffolding.get_quality_text(participant_ct_df, expanded_ct)"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "changed-aquarium",
+   "cell_type": "markdown",
+   "id": "surgical-continuity",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "assert len(expanded_ct[(expanded_ct['Mode_confirm'] == 'Pilot ebike') & (expanded_ct[\"Replaced_mode\"] == \"Pilot ebike\")]) == 0"
+    "## Metrics for Specific Mode"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "every-stanley",
+   "id": "romance-green",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Energy Impact Calculation\n",
-    "scaffolding.unit_conversions(expanded_ct)"
+    "data_eb = expanded_ct.query(f\"Mode_confirm == '{mode_of_interest}'\")"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "empty-intensity",
+   "id": "seeing-court",
    "metadata": {},
    "outputs": [],
    "source": [
-    "file_suffix = scaffolding.get_file_suffix(year, month, program)"
+    "quality_text_mode = scaffolding.get_quality_text_ebike(expanded_ct, data_eb)"
```

Comment:
does this need to be hardcoded to ebike?

Reply:
It only adds the word ebike to the quality text, I believe the generic one works fine so I removed it.